src/poller/mod.rs

```diff
@@ -165,3 +168,26 @@ where
     /// Number of generated events.
     fn poll(&mut self, timeout: Option<Duration>) -> io::Result<usize>;
 }
+
+/// Waker object provided by the poller.
+pub trait Waker {
+    /// Data tyoe for sending wake signals to the poller.
+    type Send: WakerSend;
+    /// Data type for receiving wake signals inside the poller.
+    type Recv: WakerRecv;
+
+    /// Constructs pair of waker receiver and sender objects.
+    fn pair() -> Result<(Self::Send, Self::Recv), io::Error>;
+}
+
+/// Sending part of the waker.
+pub trait WakerSend: AsRawFd + Send + Sync + Clone {
```

Comment:
I don't think the sender needs to be `AsRawFd`, does it? Since it won't be registered in a poller.